### PR TITLE
gosmee: init at 0.21.0

### PR DESCRIPTION
**Pull request by vdemeester**
## Description of changes

This adds a new package, [`gosmee`](https://github.com/chmouel/gosmee) to nixpkgs.

> Gosmee enables you to relay webhooks from either itself (as a server) or from <https://smee.io> to your local laptop or an infrastructure that is not publicly available from the internet.
>
> Gosmee let you easily expose the service on your local network (like a web service on [localhost](https://en.wikipedia.org/wiki/Localhost)) or behind a VPN, allowing a public service (such as GitHub) push webhooks into it.
>
> For instance, if you configure your GitHub Webhook to direct to a <https://smee.io/> URL or where gosmee server is listening, you can then use the gosmee client on your local notebook to obtain the events from the server and forward them to the local service, thereby establishing a connection between the GitHub webhook and your local service on your workstation.
>
> Alternatively if you don't want to use a relay server and use GitHub you can replay the hooks deliveries via the GitHub API.

This also adds @chmouel as a maintainer of this package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3356,6 +3356,12 @@
     githubId = 1118859;
     name = "Scott Worley";
   };
+  chmouel = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3359]: This change should be in an extra commit named ``maintainers: add chmouel``.

File: pkgs/by-name/go/gosmee/package.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, installShellFiles
+}:
+
+buildGoModule rec {
+  name = "gosmee-${version}";
+  version = "0.21.0";
+  rev = "v${version}";
```
Comment (SuperSandro2000) [inline pkgs/by-name/go/gosmee/package.nix:10]:
```suggestion
  pname = "gosmee";
  version = "0.21.0";
```

File: pkgs/by-name/go/gosmee/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  src = fetchFromGitHub {
+    inherit rev;
+    owner = "chmouel";
+    repo = "gosmee";
```
Comment (SuperSandro2000) [inline pkgs/by-name/go/gosmee/package.nix:15]:
```suggestion
    owner = "chmouel";
    repo = "gosmee";
    rev = "v${version}"
```

File: pkgs/by-name/go/gosmee/package.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    sha256 = "sha256-VNBz6roula/YlsYMM1kjQT3TLuQ86/MESzNNkEf/92M=";
+  };
+  vendorHash = null;
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postUnpack = ''
+    printf ${version} > $sourceRoot/gosmee/templates/version
+  '';
```
Comment (SuperSandro2000) [inline pkgs/by-name/go/gosmee/package.nix:24]:
```suggestion
  postPatch = ''
    echo ${version} > gosmee/templates/version
  '';
```
changing files should be done in patchPhase

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/go/gosmee/package.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, installShellFiles
+}:
+
+buildGoModule rec {
+  name = "gosmee-${version}";
```
Comment (SuperSandro2000) [inline pkgs/by-name/go/gosmee/package.nix:8]:
```suggestion
  pname = "gosmee";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by NixHelp** [comment]
Could we get this into [NixOS 24.05](https://github.com/NixOS/nixpkgs/issues/303286)? It would be really helpful since the [npm module](https://www.npmjs.com/package/smee-client) is not in nixpkgs either.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/go/gosmee/package.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    repo = "gosmee";
+    rev = "v${version}";
+    sha256 = "sha256-VNBz6roula/YlsYMM1kjQT3TLuQ86/MESzNNkEf/92M=";
+  };
+  vendorHash = null;
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postPatch = ''
+    printf ${version} > $sourceRoot/gosmee/templates/version
```
Comment (SuperSandro2000) [inline pkgs/by-name/go/gosmee/package.nix:22]:
```suggestion
    mkdir -p $sourceRoot/gosmee/templates
    printf ${version} > $sourceRoot/gosmee/templates/version
```

```
source root is source
Running phase: patchPhase
/nix/store/wq046wv3s3h4m2n9ak3j534di2pzra93-stdenv-linux/setup: line 114: source/gosmee/templates/version: No such file or directory
/nix/store/wq046wv3s3h4m2n9ak3j534di2pzra93-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/l759i5krlyliwbk7lr0jdg3cj4qn0rnf-gosmee-0.21.0.drv' failed with exit code 1;
```

Comment (vdemeester) [reply pkgs/by-name/go/gosmee/package.nix:22]: in `postPatch`, `PWD` already points to `/build/source` and `sourceRoot` points to `source`, so it fails anyway.

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Pull request merged**